hcipy/optics/segmented_mirror.py

```diff
@@ -0,0 +1,227 @@
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .optical_element import OpticalElement
+from ..aperture import circular_aperture, hexagonal_aperture, make_segmented_aperture
+from ..field import Field, make_hexagonal_grid, make_pupil_grid, evaluate_supersampled
+from ..io import write_fits
+from ..plotting import imshow_field
+
+PUP_DIAMETER = 15.   # m
+
+def get_atlast_aperture(normalized=False, with_segment_gaps=True, segment_transmissions=1, return_segment_positions=False):
+	"""Make the ATLAST/HiCAT pupil mask.
+
+	This function is a copy of make_hicat_aperture(), except that it also returns the segment positions.
+
+	Parameters
+	----------
+	normalized : boolean
+		If this is True, the outer diameter will be scaled to 1. Otherwise, the
+		diameter of the pupil will be 15.0 meters.
+	with_segment_gaps : boolean
+		Include the gaps between individual segments in the aperture.
+	segment_transmissions : scalar or array_like
+		The transmission for each of the segments. If this is a scalar, this transmission will
+		be used for all segments.
+	return_segment_positions : boolean
+		If this is True, the centers of each of the segments will get returned as well.
+	
+	Returns
+	-------
+	Field generator
+		The ATLAST aperture.
+	CartesianGrid
+		The segment positions. Only returned when `return_segment_positions` is True.
+	"""
+	pupil_diameter = 15 # m
+	segment_circum_diameter = 2 / np.sqrt(3) * pupil_diameter / 7
+	num_rings = 3
+	segment_gap = 0.01 # m
+
+	if not with_segment_gaps:
+		segment_gap = 0
+
+	if normalized:
+		segment_circum_diameter /= pupil_diameter
+		segment_gap /= pupil_diameter
+		pupil_diameter = 1.0
+
+	segment_positions = make_hexagonal_grid(segment_circum_diameter / 2 * np.sqrt(3), num_rings)
+	segment_positions = segment_positions.subset(lambda grid: ~(circular_aperture(segment_circum_diameter)(grid) > 0))
+
+	hexagon = hexagonal_aperture(segment_circum_diameter - segment_gap, np.pi / 2)
+
+	def segment(grid):
+		return hexagon(grid)
```

Comment:
The same thing can be done for the HiCAT and LUVOIR aperture. Should we include that in this PR?

Reply:
I added these changes to the HiCAT and LUVOIR apertures in commit 7dde6f31ef55bb5d4bc40c415d8a5813bd804814.